ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -283,6 +284,19 @@ func (r *RayClusterReconciler) rayClusterReconcile(ctx context.Context, request
 							"We do not need to requeue the RayCluster CR anymore.", instance.Name))
 					return ctrl.Result{}, nil
 				}
+				// redisCleanupJob.Spec.ActiveDeadlineSeconds is set by the GCS_FT_REDIS_CLEANUP_DEADLINE_SECONDS env
+				if redisCleanupJob.Spec.ActiveDeadlineSeconds != nil && utils.IsJobFinished(&redisCleanupJob) {
```

Comment:
(You can ignore this comment if we decide to merge the logic.)
Here, we may only need to check for `batchv1.JobFailed`. The K8s Job is a non-parallel Job, so the Job is complete as soon as its Pod terminates successfully ([ref](https://kubernetes.io/docs/concepts/workloads/controllers/job/#parallel-jobs)).